likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js

```diff
@@ -112,24 +117,23 @@ function LikecoinButtonPage() {
     e.preventDefault();
     setShowChangeButton(false);
     const data = {
-      userLikerInfos: {
-        likecoin_id:
-          likerDisplayName === '-' ? DBUserLikerId : likerIdValue,
-        display_name:
-          likerDisplayName === '-'
-            ? DBUserLikerDisplayName
-            : likerDisplayName,
-        wallet:
-          likerDisplayName === '-'
-            ? DBUserLikerWallet
-            : likerWalletAddress,
-        avatar:
-          likerDisplayName === '-'
-            ? DBUserLikerAvatar
-            : likerAvatar,
-      },
+      userLikerInfos: {},
     };
+    if (likerIdValue) setForceDisconnect(false);
+    if (forceDisconnect) {
+      data.userLikerInfos.likecoin_id = '';
+      data.userLikerInfos.display_name = '';
+      data.userLikerInfos.wallet = '';
+      data.userLikerInfos.avatar = '';
+    } else {
+      data.userLikerInfos.likecoin_id = likerDisplayName === '-' ? DBUserLikerId : likerIdValue;
+      data.userLikerInfos.display_name = likerDisplayName === '-' ? DBUserLikerDisplayName : likerDisplayName;
+      data.userLikerInfos.wallet = likerDisplayName === '-' ? DBUserLikerWallet : likerWalletAddress;
+      data.userLikerInfos.avatar = likerDisplayName === '-' ? DBUserLikerAvatar : likerAvatar;
```

Comment:
I think a much better way is to rewrite this to split the display data (`-` for empty values) and actual data (`''` or `null` for empty values) in local state of this component.
In such design you no need to handle `-` as a data value or add `forceDisconnect` as a hacky flag, just handle empty data as disconnect would do

Reply:
Changed it in the simplify UI logic commit.